gr-filter/lib/ival_decimator_impl.cc

```diff
@@ -0,0 +1,90 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2020 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "ival_decimator_impl.h"
+#include <gnuradio/io_signature.h>
+
+namespace gr {
+namespace filter {
+
+
+ival_decimator::sptr ival_decimator::make(int decimation, int data_size)
+{
+    return gnuradio::get_initial_sptr(new ival_decimator_impl(decimation, data_size));
+}
+
+/*
+ * The private constructor
+ */
+ival_decimator_impl::ival_decimator_impl(int decimation, int data_size)
+    : gr::sync_decimator("ival_decimator",
+                         gr::io_signature::make(1, 1, data_size),
+                         gr::io_signature::make(1, 1, data_size),
+                         decimation),
+      d_data_size(data_size)
+{
+    d_doubledecimation = 2 * decimation;
+
```

Comment:
please add a check for `data_size` to be either `sizeof(char)` or `sizeof(short)` ... or throw an error here during instantiation.